src/consumer.rs

```diff
@@ -76,12 +76,26 @@ impl ConsumerBuilder {
                     metadata.replicas,
                     stream
                 );
-                client = Client::connect(ClientOptions {
-                    host: replica.host.clone(),
-                    port: replica.port as u16,
-                    ..self.environment.options.client_options
-                })
-                .await?;
+                let load_balancer_mode = self.environment.options.client_options.load_balancer_mode;
+                if load_balancer_mode {
+                    let options = self.environment.options.client_options.clone();
+                    loop {
+                        let temp_client = Client::connect(options.clone()).await?;
+                        let mapping = temp_client.connection_properties().await;
+                        let advertised_host = mapping.get("advertised_host").unwrap();
```

Comment:
Please remove the `unwrap`, if the property `advertised_host` is not found return an error. 
Same for the producer